### PR TITLE
[python] Update the `MultiscaleImage` to match `somacore` API changes 

### PR DESCRIPTION
**Pull request by jp-dark**
This is the companion TileDB-SOMA changes for single-cell-data/SOMA#238. It consists of major changes to the `MultiscaleImage` implementation.

Do NOT merge before single-cell-data/SOMA#238.

### CODE REVIEW

File: apis/python/setup.py [hunk 1]
```diff
@@ -335,8 +335,8 @@ def run(self):
         "pyarrow",
         "scanpy>=1.9.2",
         "scipy",
-        # Note: the somacore version is also pinned in .pre-commit-config.yaml
-        "somacore==1.0.22",
+        # Note: the somacore version is in .pre-commit-config.yaml too
+        "somacore @ git+https://github.com/single-cell-data/SOMA.git@08b795e",  # DO NOT MERGE
```
Comment (jp-dark) [inline apis/python/setup.py:339]: Do NOT merge until this is updated.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ repos:
         # Pandas 2.x types (e.g. `pd.Series[Any]`). See `_types.py` or https://github.com/single-cell-data/TileDB-SOMA/issues/2839
         # for more info.
         - "pandas-stubs>=2"
-        - "somacore==1.0.22"
```
Comment (jp-dark) [inline .pre-commit-config.yaml:21]: Do NOT merge until this is updated.

**Review by aaronwolen — APPROVED** [review]
🚢

File: apis/python/setup.py (hunk 1 above)
Comment (jp-dark) [reply apis/python/setup.py]: Merging to new release branch instead of main.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jp-dark) [reply .pre-commit-config.yaml]: Merging to new release branch instead of main.

**Pull request merged**